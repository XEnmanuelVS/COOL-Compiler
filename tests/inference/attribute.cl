class Point {
    x: AUTO_TYPE;
    y: AUTO_TYPE;
    z: AUTO_TYPE;

    init(n: Int, m: Int): AUTO_TYPE {
        {
            z <- x;
            x <- n;
            y <- m;
        }
    };
};
