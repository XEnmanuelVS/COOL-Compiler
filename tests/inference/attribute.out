\__ProgramNode [<class> ... <class>]
  \__ClassDeclarationNode: class Point  { <feature> ... <feature> }
    \__AttrDeclarationNode: x : Int
    \__AttrDeclarationNode: y : Int
    \__AttrDeclarationNode: z : Int
    \__FuncDeclarationNode: init(n: Int, m: Int) : Int -> <body>
      \__BlockNode:
        \__AssignNode: z <- <expr>
          \__VariableNode: x
        \__AssignNode: x <- <expr>
          \__VariableNode: n
        \__AssignNode: y <- <expr>
          \__VariableNode: m

